src/lib.rs

```diff
@@ -341,6 +341,57 @@ impl<const LEN: usize> AStr<LEN> {
         };
         AStr::<RET_LEN>::from_utf8_array_unchecked(ret_buf)
     }
+
+    pub fn try_from_fmt(display: impl core::fmt::Display) -> Result<Self, core::fmt::Error> {
+        use core::fmt::Write;
+        let mut builder = FmtBuilder::new();
+        write!(builder, "{}", display)?;
+        builder.finalize()
+    }
+}
+
+/// Private type to build an [`AStr`] from anything that can print to an [core::fmt::Write]
+struct FmtBuilder<const LEN: usize> {
```

Comment:
maybe rename the `const` parameter `LEN` to `CAP` or `CAPACITY` 
so it would be clearer what the difference ist to `len`

Reply:
You're 100% right, that's a relic from when I was experimenting.